File: scripts/py-compile-suite/inputs/30_full_pipeline.m
function [peaks, locs] = findpeaks_demo(signal, fs, threshold)
    % Find local peaks above a threshold
    N = length(signal);
    dt = 1/fs;
    t = (0:N-1) * dt;

    % Smooth with a lowpass filter
    [b, a] = butter(4, 0.3);
    filtered = filtfilt(b, a, signal);

    % Scan for local maxima above threshold
    peaks = [];
    locs = [];
    for i = 2:N-1
        if filtered(i) > filtered(i-1) && filtered(i) > filtered(i+1)
            if filtered(i) > threshold
                peaks = [peaks, filtered(i)];
                locs = [locs, t(i)];
            end
        end
    end

    % Plot the result
    figure;
    plot(t, signal, 'b', 'LineWidth', 0.5);
    hold on;
    plot(locs, peaks, 'rv', 'MarkerSize', 8);
    title(sprintf('Found %d peaks above %.2f', length(peaks), threshold));
    xlabel('Time (s)');
    ylabel('Amplitude');
    grid on;
    legend('Signal', 'Peaks', 'Location', 'Best');
end

File: scripts/py-compile-suite/outputs/30_full_pipeline.py
import numpy as np
import scipy.signal as signal
import matplotlib.pyplot as plt

def findpeaks_demo(signal_, fs, threshold):
    # Find local peaks above a threshold
    N = np.max(signal_.shape)
    dt = 1/fs
    t = np.arange(0, N-1 + 1) * dt

    # Smooth with a lowpass filter
    b, a = signal.butter(4, 0.3)
    filtered = signal.filtfilt(b, a, signal_)

    # Scan for local maxima above threshold
    peaks = []
    locs = []
    for i in range(2, N-1 + 1):
        if filtered[i - 1] > filtered[i-1] and filtered[i - 1] > filtered[i+1]:
            if filtered[i - 1] > threshold:
                peaks = np.append(peaks, filtered[i - 1])
                locs = np.append(locs, t[i - 1])

    # Plot the result
    plt.figure()
    plt.plot(t, signal_, 'b', linewidth=0.5)
    # hold on removed — matplotlib accumulates plots by default
    plt.plot(locs, peaks, 'rv', markersize=8)
    plt.title('Found %d peaks above %.2f' % (np.max(peaks.shape), threshold))
    plt.xlabel('Time (s)')
    plt.ylabel('Amplitude')
    plt.grid(True)
    plt.legend('Signal' + 'Peaks', loc='best')
    return peaks, locs
